docs/src/jump_types.md

```diff
@@ -26,116 +27,111 @@ a stochastic differential equation to have jumps is commonly known as a Jump
+dependent on values which may change between the occurrence of any two jump
+events of the process. Examples include jumps where the rate is an explicit
+function of time, or depends on a state variable that is modified via continuous
+dynamics such as an ODE or SDE. Such "general" `VariableRateJump`s can be
+expensive to simulate because it is necessary to take into account the (possibly
+continuous) changes in the rate function when calculating the next jump time.
+
+*Bounded* [`VariableRateJump`](@ref)s represent a special subset of
+`VariableRateJump`s where one can specify functions that calculate a time window
+over which the rate is bounded by a constant (presuming the state `u` is
+unchanged due to another `ConstantRateJump`, `MassActionJump` or bounded
+`VariableRateJump`). They can be simulated more efficiently using
+rejection-sampling based approaches that leverage this upper bound.
+
+[`ConstantRateJump`](@ref)s are more restricted in that they assume the rate
+functions are constant at all times between two consecutive jumps of the system.
+That is, any states or parameters that a rate function depends on must not
+change between the times at which two consecutive jumps occur.
+
+A [`MassActionJump`](@ref)s is a specialized representation for a collection of
+`ConstantRateJump` jumps that can each be interpreted as a standard mass action
+reaction. For systems comprised of many mass action reactions, using the
+`MassActionJump` type will offer improved performance comnpared to using
+multiple `ConstantRateJump`s. Note, only one `MassActionJump` should be defined
+per [`JumpProblem`](@ref); it is then responsible for handling all mass action
 reaction type jumps. For systems with both mass action jumps and non-mass action
 jumps, one can create one `MassActionJump` to handle the mass action jumps, and
-create a number of `ConstantRateJumps` or `VariableRateJump` to handle the
+create a number of `ConstantRateJump`s or `VariableRateJump`s to handle the
 non-mass action jumps.
 
-`RegularJump`s are optimized for inexact jumping algorithms like tau-leaping and
-hybrid algorithms. `ConstantRateJump`, `VariableRateJump`, `MassActionJump` are
-optimized for exact methods (also known in the biochemistry literature as SSA
-algorithms). `ConstantRateJump`s, `VariableRateJump`s and `MassActionJump`s can
-be added to standard DiffEq algorithms since they are simply callbacks, while
-`RegularJump`s require special algorithms.
+Since exact methods simulate each individual jump, they may become
+computationally expensive to simulate processes over timescales that involve
+*many* jump occurrences. As an alternative, inexact τ-leaping methods take
+discrete steps through time, over which they simultaneously execute many jumps.
+These methods can be much faster as they do not need to simulate the realization
+of every individual jump event. τ-leaping methods trade accuracy for speed, and
```

Comment:
I think it would make sense to add some tau-leap examples to the first and second tutorials, and we can mention this there. It isn’t well documented,meted currently anyways…

Reply:
I’m going to leave that for a follow up PR.